### PR TITLE
[WIP] Upgrade to Core 25.2.2, includes test upgrades

### PR DESCRIPTION
**Pull request by verbman**
* Technical improvement.
* Impacted periods: all. 
* Impacted areas: `tests, OpenFisca.Core`
* Details:
  - Upgrade to the new OpenFisca.Core 25.2.2

- - - -

These changes:

- Impact the OpenFisca-Aotearoa public API installation
- Change non-functional parts of this repository (for instance editing the README)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,13 +16,13 @@
     url='https://github.com/ServiceInnovationLab/openfisca-aotearoa',
     include_package_data=True,  # Will read MANIFEST.in
     install_requires=[
-        'OpenFisca-Core >= 24.3.0, < 25.0',
+        'OpenFisca-Core >= 25.2.2, < 26.0',
```
Comment (Br3nda) [inline setup.py]: which version was DAY support added in ?

**Comment by Br3nda** [comment]
Heroku loads dependencies from requirements.txt, so that file also needs updating.
Right now it loads openfisca from git

**Comment by Br3nda** [comment]
Are you still working on this? Should we close it or mark it as WIP?

**Comment by verbman** [comment]
Happy to finish it but am now otherwise occupied till the end of the month.
If you need the upgrade before then you're welcome to finish it up yourself.
Yes it has DAY.

**Comment by Br3nda** [comment]
I've marked it as WIP so the bots don't remind us to progress it every day

**Comment by Br3nda** [comment]
replaced with #145

**Pull request closed without merging**